AG-9272 Error Bar Highlight Style

olegat opened the pull request:
https://ag-grid.atlassian.net/browse/AG-9272

olegat committed:
AG-9272 Implement errorbar highlight style

olegat commented:
This has a merge conflict with https://github.com/ag-grid/ag-charts/pull/351 and may need a rebase

alantreadway commented on packages/ag-charts-community/src/chart/series/cartesian/cartesianSeries.ts:
```diff
@@ -47,6 +47,9 @@ export interface ErrorBoundSeriesNodeDatum {
     readonly capDefaults: { lengthRatio: number; lengthRatioMultiplier: number; lengthMax: number };
+    // The error bar module needs to know which node data to retrieve when
+    // events on are triggered on a datum.
+    readonly index: number;
```
Can we discuss the use of `index` for this use case - we haven't needed it for any other cases, so I'm a little suspicious that this is needed?

olegat replied:
The `index` is required for the `ErrorBar` class to know which `ErrorBarNode` scene object to update.

However, I've removed this property. Instead, the `ErrorBar.restyleHightlightChange` method will determine the index by searching sequentially for a matching `datum` object.

olegat committed:
AG-9272 Remove ErrorBoundSeriesNodeDatum.index

As a substitute, search sequentially for the matching `ErrorBarNode`
instance in the new method `ErrorBar.restyleHightlightChange`. This isn't
a good solution in terms of performance. However, the assumption right
now is that the typical use case for error bars includes few data points
(because the chart will get cluttered very quickly if there are many
data points with error bars). This node lookup can be revisited if
performance becomes an issue.

Also separate the style and translation updates in `ErrorBarNode`. This
allows use to specify whether and major or minor scene change occurs.

alantreadway commented on packages/ag-charts-community/src/chart/series/cartesian/scatterSeries.ts:
```diff
@@ -170,7 +170,8 @@ export class ScatterSeries extends CartesianSeries<Group, ScatterNodeDatum> {
         const font = label.getFont();
-        for (const { values, datum } of processedData.data ?? []) {
+        for (let i = 0; i < processedData.data.length; i++) {
+            const { values, datum } = processedData.data[i];
```
Nit: Can we just revert this line to remove the use of `i`?

olegat replied:
Well spotted. Yes of course. I'll revert this.

alantreadway reviewed (approved):
LGTM other than one nit - thanks for the updates :-)

olegat committed:
Revert use of index in ScatterSeries.createNodeData
